### DISCUSSION TITLE
How to use a query in new modular firestore

### DISCUSSION THREAD

**Discussion by m8xp0w3r** [angular/angularfire#3203] [Q&A] [answered]
Hello everyone,
I am moving from compat to modular firestore version.  I have the following function with an optional query argument:

```javascript
getCollection<T>(collectionName: string, queryFn?: QueryFn): AngularFirestoreCollection<T> {
    return this.firestore.collection<T>(collectionName, queryFn);
  }
```

With the modular version I probably will go like this:
```javascript
getCollection(collectionName: string, queryFn?: QueryFn): CollectionReference<DocumentData> {
    return collection(this.firestore, collectionName);
  }
```

Unfortunately I did not find anything how to pass my (optional) Query here. Do you have any suggestions how I I can do this?

Regards Nils

**Comment by gustavo-alarcon** [comment, marked as answer]
Hi there !

I use the following approach, perhaps it could help you.

```js
getUsers(): Observable<User[]> {
    const usersCollection: CollectionReference<User> = collection(
      this.firestore,
      'users'
    ) as CollectionReference<User>;

    return collectionData<User>(
      query(usersCollection, orderBy('createdAt', 'desc')),
      {
        idField: 'uid',
      }
    );
  }
```

Have a good day buddy !

**Reply by m8xp0w3r** [reply]
Awesome. This helped me a lot. My biggest problem was, that my IDE (PHP Storm) did not import all functions and types correct (some were imported from compat, others were not automatically detected).
